libraries/chain/proposal_evaluator.cpp

```diff
@@ -190,54 +190,54 @@ void_result proposal_create_evaluator::do_evaluate(const proposal_create_operati
+   FC_ASSERT (o.expiration_time <= block_time + global_parameters.maximum_proposal_lifetime,
               "Proposal expiration time is too far in the future.");
-   FC_ASSERT( !o.review_period_seconds || fc::seconds(*o.review_period_seconds) < (o.expiration_time - block_time),
-              "Proposal review period must be less than its overall lifetime." );
-
-   {
-      // If we're dealing with the committee authority, make sure this transaction has a sufficient review period.
-      flat_set<account_id_type> auths;
-      vector<authority> other;
-      for( auto& op : o.proposed_ops )
-      {
-         operation_get_required_authorities(op.op, auths, auths, other);
-      }
-
-      FC_ASSERT( other.size() == 0 ); // TODO: what about other??? 
-
-      if( auths.find(GRAPHENE_COMMITTEE_ACCOUNT) != auths.end() )
-      {
-         GRAPHENE_ASSERT(
-            o.review_period_seconds.valid(),
-            proposal_create_review_period_required,
-            "Review period not given, but at least ${min} required",
-            ("min", global_parameters.committee_proposal_review_period)
-         );
-         GRAPHENE_ASSERT(
-            *o.review_period_seconds >= global_parameters.committee_proposal_review_period,
-            proposal_create_review_period_insufficient,
-            "Review period of ${t} specified, but at least ${min} required",
-            ("t", *o.review_period_seconds)
-            ("min", global_parameters.committee_proposal_review_period)
-         );
-      }
+   FC_ASSERT (!o.review_period_seconds || fc::seconds(*o.review_period_seconds) < (o.expiration_time - block_time),
+              "Proposal review period must be less than its overall lifetime.");
+
+   // Find all authorities required by the proposed operations
+   flat_set<account_id_type> tmp_required_active_auths;
+   vector<authority> other;
+   for (auto& op : o.proposed_ops) {
+      operation_get_required_authorities(op.op, tmp_required_active_auths, _required_owner_auths, other,
+                                         MUST_IGNORE_CUSTOM_OP_REQD_AUTHS(block_time));
+   }
+   // All accounts which must provide both owner and active authority should be omitted from the active authority set;
+   // owner authority approval implies active authority approval.
+   std::set_difference(tmp_required_active_auths.begin(), tmp_required_active_auths.end(),
+                       _required_owner_auths.begin(), _required_owner_auths.end(),
+                       std::inserter(_required_active_auths, _required_active_auths.begin()));
+
+   // TODO: what about other???
+   FC_ASSERT (other.empty(), "Proposals containing operations requiring non-account authorities are not yet implemented.");
+
+   // If we're dealing with the committee authority, make sure this transaction has a sufficient review period.
+   if (_required_active_auths.count(GRAPHENE_COMMITTEE_ACCOUNT) || _required_owner_auths.count(GRAPHENE_COMMITTEE_ACCOUNT)) {
```

Comment:
Note: According to documentation of `flat_set`, both `find` and `count` have logarithmic complexity. Question is which is better readable. I'm slightly leaning towards `count`.

Speaking of readability, this line is too long. I think we agreed on 123 chars?

Reply:
123 is a good number; I'll wrap the lines.

I find `count` to be more readable, especially because it's far less verbose.